ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/inspection_service.proto

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v1.admin;
+
+import "com/daml/ledger/api/v1/value.proto";
+import "com/daml/ledger/api/v1/event.proto";
+
+option java_outer_classname = "InspectionServiceOuterClass";
+option java_package = "com.daml.ledger.api.v1.admin";
+option csharp_namespace = "Com.Daml.Ledger.Api.V1.Admin";
+
+// Experimental API allow participant inspection
+service InspectionService {
+    rpc GetContractById (GetContractByIdRequest) returns (GetContractByIdResponse);
```

Comment:
This API reminds of a point-wise ACS service. 
Would it make sense to extend the Inspection Service such the its requests include `verbose` and `active_at_offset` fields; and its responses include `offset` and `workflow_id` fields, with the same semantics as in ACS service?

What is the behavior of `GetContractById` and `GetContractByKey` if the contract doesn't exists? What if the contract exists but was archived? 

Would it be useful to have an ability to return a history of contract create and archive events per contractId/contractKey?

Reply:
I like the direction in which @pbatko-da suggestions are going. However, I think it is more interesting to explore the similarities to the transaction service rather than the ACS.

- Returning both create and archive events is very useful for troubleshooting and can be easily collapsed into one by the client code
- Returning a list would also make this code robust to introduction of "non-unique" contract keys in the future
- `verbose` and `active_at_offset` all make sense
- i would also add the ability to return interface projections and create_blobs so that the result can be used for explicit disclosure

Given all the above I would seriously consider moving the calls to the transaction service proper